# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1300568018 25200
#      Sat Mar 19 13:53:38 2011 -0700
# Node ID d70c99028ba33e5a6d9de8b55d34648cef8730dc
# Parent  0579a13f29a1c3fceb347b8fae04793273a087f7
Make helper functions for logistic_regression private.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,17 @@
+2010-03-19  Rik  <octave@nomad.inbox5.com>
+
+	* statistics/models/logistic_regression.m: Do not split function
+	declaration with line continuation.
+	* statistics/models/private/logistic_regression_likelihood.m,
+	statistics/models/private/logistic_regression_derivatives.m: Make
+	helper functions private.
+	* statistics/models/module.mk: Make helper functions private.
+
 2010-03-19  Rik  <octave@nomad.inbox5.com>
 
 	* plot/isocolors.m, plot/isonormals.m, plot/isosurface.m: Improve
 	docstrings.
 
 2010-03-19  Rik  <octave@nomad.inbox5.com>
 
 	* miscellaneous/dump_prefs.m: Close @deftypefn macro left open.
diff --git a/scripts/statistics/models/logistic_regression.m b/scripts/statistics/models/logistic_regression.m
--- a/scripts/statistics/models/logistic_regression.m
+++ b/scripts/statistics/models/logistic_regression.m
@@ -79,18 +79,17 @@
 
 ## Author: Gordon K Smyth <gks@maths.uq.oz.au>,
 ## Adapted-By: KH <Kurt.Hornik@wu-wien.ac.at>
 ## Description: Ordinal logistic regression
 
 ## Uses the auxiliary functions logistic_regression_derivatives and
 ## logistic_regression_likelihood.
 
-function [theta, beta, dev, dl, d2l, p] ...
-  = logistic_regression (y, x, print, theta, beta)
+function [theta, beta, dev, dl, d2l, p] = logistic_regression (y, x, print, theta, beta)
 
   ## check input
   y = round (vec (y));
   [my, ny] = size (y);
   if (nargin < 2)
     x = zeros (my, 0);
   endif;
   [mx, nx] = size (x);
diff --git a/scripts/statistics/models/module.mk b/scripts/statistics/models/module.mk
--- a/scripts/statistics/models/module.mk
+++ b/scripts/statistics/models/module.mk
@@ -1,12 +1,15 @@
 FCN_FILE_DIRS += statistics/models
 
+statistics_models_PRIVATE_FCN_FILES = \
+  statistics/models/private/logistic_regression_derivatives.m \
+  statistics/models/private/logistic_regression_likelihood.m
+
 statistics_models_FCN_FILES = \
   statistics/models/logistic_regression.m \
-  statistics/models/logistic_regression_derivatives.m \
-  statistics/models/logistic_regression_likelihood.m
+  $(statistics_models_PRIVATE_FCN_FILES) 
 
 FCN_FILES += $(statistics_models_FCN_FILES)
 
 PKG_ADD_FILES += statistics/models/PKG_ADD
 
 DIRSTAMP_FILES += statistics/models/$(octave_dirstamp)
diff --git a/scripts/statistics/models/logistic_regression_derivatives.m b/scripts/statistics/models/private/logistic_regression_derivatives.m
rename from scripts/statistics/models/logistic_regression_derivatives.m
rename to scripts/statistics/models/private/logistic_regression_derivatives.m
--- a/scripts/statistics/models/logistic_regression_derivatives.m
+++ b/scripts/statistics/models/private/logistic_regression_derivatives.m
@@ -13,18 +13,20 @@
 ## General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {[@var{dl}, @var{d2l}] =} logistic_regression_derivatives (@var{x}, @var{z}, @var{z1}, @var{g}, @var{g1}, @var{p})
-## Called by logistic_regression.  Calculates derivates of the
-## log-likelihood for ordinal logistic regression model.
+## Calculate derivatives of the log-likelihood for ordinal logistic regression
+## model.
+## Private function called by @code{logistic_regression}.
+## @seealso{logistic_regression}
 ## @end deftypefn
 
 ## Author: Gordon K. Smyth <gks@maths.uq.oz.au>
 ## Adapted-By: KH <Kurt.Hornik@wu-wien.ac.at>
 ## Description: Derivates of log-likelihood in logistic regression
 
 function [dl, d2l] = logistic_regression_derivatives (x, z, z1, g, g1, p)
 
diff --git a/scripts/statistics/models/logistic_regression_likelihood.m b/scripts/statistics/models/private/logistic_regression_likelihood.m
rename from scripts/statistics/models/logistic_regression_likelihood.m
rename to scripts/statistics/models/private/logistic_regression_likelihood.m
--- a/scripts/statistics/models/logistic_regression_likelihood.m
+++ b/scripts/statistics/models/private/logistic_regression_likelihood.m
@@ -13,18 +13,19 @@
 ## General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {[@var{g}, @var{g1}, @var{p}, @var{dev}] =} logistic_regression_likelihood (@var{y}, @var{x}, @var{beta}, @var{z}, @var{z1})
-## Calculates likelihood for the ordinal logistic regression model.
-## Called by logistic_regression.
+## Calculate the likelihood for the ordinal logistic regression model.
+## Private function called by @code{logistic_regression}.
+## @seealso{logistic_regression}
 ## @end deftypefn
 
 ## Author: Gordon K. Smyth <gks@maths.uq.oz.au>
 ## Adapted-By: KH <Kurt.Hornik@wu-wien.ac.at>
 ## Description: Likelihood in logistic regression
 
 function [g, g1, p, dev] = logistic_regression_likelihood (y, x, beta, z, z1)
 
